### PR TITLE
traefik mesh lab

### LINKED ISSUE #29: Lab for `Traefik Mesh`
**Current Behavior**

Currently we don't have scenario `Traefik Mesh` under `Meshery-Adapters`

**Desired Behavior**

Create a new scenario under `Meshery-Adapters` for `Traefik Mesh`

---
**Resources**

[Sample Lab](https://github.com/layer5io/service-mesh-labs/tree/master/Meshery-Adapters/consul-meshery-adapter)

**Alternatives / Additional Context**

### PR DESCRIPTION
**Pull request by warunicorn19**
Signed-off-by: Aditya Chaterjee <speak2adi@gmail.com>

**Description**
Lab for traefik mesh adapter
This PR fixes #29 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by warunicorn19** [comment]
@suhaniii wasn't able to test locally, if you could please tell if i did anything wrong.

File: Meshery-Adapters/traefik-meshery-adapter/index.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      "hideHiddenFiles": true,
+      "exclusionPatterns": ["*test", "./test/*", "./logs"],
+      "showdashboard": true,
+      "dashboards": [{"name": "Meshery Server", "port": 9081},{"name": "Emojivoto", "port": 8080}],
+      "uilayout": "terminal-iframe"
+    },
+    "backend": {
+      "imageid": "kubernetes-cluster-running"
+    }
+  }
```
Comment (suhaniii) [inline Meshery-Adapters/traefik-meshery-adapter/index.json:47]: @warunicorn19  have you tested `kubernetes-cluster-running` image-id?

File: Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+/root/shell.sh
```
Comment (suhaniii) [inline Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh:3]: You need to think about `/root/shell.sh`, with `minikube` as image id we have to start Minikube using `/root/shell.sh`, which we don't want.
Now with `kubernetes-cluster-running` you need to identify if  `/root/shell.sh` is required.

**Comment by leecalcote** [comment]
Hello! It's nice to see this PR.

File: Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh (hunk 2 above)
Comment (warunicorn19) [reply Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh:3]: Alright, I'll have a look

File: Meshery-Adapters/traefik-meshery-adapter/index.json (hunk 1 above)
Comment (warunicorn19) [reply Meshery-Adapters/traefik-meshery-adapter/index.json:47]: Haven't tested it, but I thought that's what we required after yesterday's call .

Comment (suhaniii) [reply Meshery-Adapters/traefik-meshery-adapter/index.json:47]: Yep we do require Kubernetes running, we have to make sure that it compatible with existing instructions.

**Comment by suhaniii** [comment]
> Hello! It's nice to see this PR.

@warunicorn19  - Gallant

File: Meshery-Adapters/traefik-meshery-adapter/index.json (hunk 1 above)
Comment (suhaniii) [reply Meshery-Adapters/traefik-meshery-adapter/index.json:47]: @warunicorn19  what's your conclusion about `kubernetes-cluster-running` image?

**Comment by warunicorn19** [comment]
@suhaniii I tested it out locally and `Kubernetes-cluster-running` doesn't seem to work. I tried doing it with `/launch.sh` as well but still wasn't working. So, I changed it back to minikube. If there are no issues with this, let's get this and merged and look for a `Kubernetes running`  image_id.

File: Meshery-Adapters/traefik-meshery-adapter/step1.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+This session will auto-install **`minikube`** and set up other environment variables, necessary to run Meshery for you. This may take up to 2 minutes. 
+
+After the environment has been setup, you may check up on your cluster nodes by running:
+`kubectl get nodes`{{execute}}
+
+## Set up Meshery
+
+Meshery can be downloaded, installed, deployed and auto-launched with a single command:
+
+`curl -L https://git.io/meshery | PLATFORM=docker bash -`{{execute}}
```
Comment (suhaniii) [inline Meshery-Adapters/traefik-meshery-adapter/step1.md:10]:
```suggestion
`curl -L https://git.io/meshery | ADAPTERS=traefik  PLATFORM=docker bash -`{{execute}}
```

**Comment by suhaniii** [comment]
> @suhaniii I tested it out locally and `Kubernetes-cluster-running` doesn't seem to work. I tried doing it with `/launch.sh` as well but still wasn't working. So, I changed it back to minikube. If there are no issues with this, let's get this and merged and look for a `Kubernetes running` image_id.

@warunicorn19   We now have this error  -> `Minikube Dashboard is not supported via the interactive terminal experience.`
Screenshot 2021-08-30 at 9 37 40 PM

**Comment by warunicorn19** [comment]
@suhaniii is this happening only in this lab, or the other too?

**Comment by warunicorn19** [comment]
There's an open issue on this already right?  #31

**Comment by suhaniii** [comment]
> @suhaniii is this happening only in this lab, or the other too?

Others as well

**Comment by suhaniii** [comment]
@warunicorn19  Have you tried `launch.sh` for `kubernetes-cluster-running` this [way](https://github.com/katacoda/scenario-examples/blob/main/environment-usages/kubernetes-cluster-running/step1.md) ?

**Comment by warunicorn19** [comment]
@suhaniii no, i actually put changed it in `step1-foreground` and in the .json file. I'll try once like this, but I don't know if that'll work or not. i was getting the error `unable to find "launch.sh"`

**Comment by warunicorn19** [comment]
@suhaniii could you please try once? even with the way you that you had sent. maybe I'm doing something wrong.

**Comment by warunicorn19** [comment]
Also, if you could check point no. 1.6 in [this](https://www.oreilly.com/library/view/kubernetes-cookbook/9781491979679/ch01.html) that might help with the minikube dashboard problem.

**Comment by suhaniii** [comment]
> @suhaniii could you please try once? even with the way you that you had sent. maybe I'm doing something wrong.

@warunicorn19  we have some progress with `kubernetes-cluster-running`, you need to remove `/root/shell.sh` there and when you do `launch.sh` it says -> Kubernetes started.
Are you pointing to something else?

Screenshot 2021-08-31 at 5 57 33 AM

**Comment by warunicorn19** [comment]
Ahh my bad, my katacoda was not getting updated with my latest push, and I was trying to do launch.sh, so it kept showing couldn't find launch.sh

**Comment by warunicorn19** [comment]
Screenshot 2021-08-31 at 11 25 03 AM

This is what we want right? @suhaniii 

i will update the commands in that way and commit then.

File: Meshery-Adapters/traefik-meshery-adapter/step3.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+![Traefik Mesh Sample Apps](./assets/traefik-sample.png)
+
+You will be notified of successful deployment:
+
+![Traefik Mesh Imagehub](./assets/image-hub.png)
+
+### Expose your Sample Application
+
+A sidecar injector is used for automating the injection of the Linkerd proxy into your application's pod spec. The Kubernetes admission controller enforces this behavior by sending a webhook request to the sidecar injector every time a pod is to be scheduled.
```
Comment (sayantan1413) [inline Meshery-Adapters/traefik-meshery-adapter/step3.md:20]: I guess we do not have a sidecar Injector in Traefik mesh . @utkarsh-pro ?

File: Meshery-Adapters/traefik-meshery-adapter/step3.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+You have already deployed the sidecar proxy injector when you installed Consul, which should be running in your control plane. To verify, execute this command:
+
+`kubectl get deployment consul-consul-connect-injector-webhook-deployment -n consul-system`{{execute}}
```
Comment (sayantan1413) [inline Meshery-Adapters/traefik-meshery-adapter/step3.md:24]: Why are we using Consul here ?

File: Meshery-Adapters/traefik-meshery-adapter/step2.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+## 2. In the Traefik Mesh management page:
+
+- Type `traefik-system` into the namespace field
+- Click the (+) icon on the Install card and select `Traefik Mesh` to install the latest version of Traefik Mesh
+
+![Install Traefik-Mesh using Meshery](./assets/install-traefik.png)
+
+## Check for successful installation
+
+NSM has been deployed in a separate Kubernetes namespace: nsm-system. To check if NSM is along with all the pieces that have been deployed, execute the following:
```
Comment (sayantan1413) [inline Meshery-Adapters/traefik-meshery-adapter/step2.md:16]: Why are we using NSM here ?

File: Meshery-Adapters/traefik-meshery-adapter/step2.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+Meshery allows you to install any service mesh with a single click:
+
+## 1. Select `Traefik-mesh` from the Management menu
```
Comment (sayantan1413) [inline Meshery-Adapters/traefik-meshery-adapter/step2.md:3]:
```suggestion
## 1. Select `Traefik-Mesh` from the Management menu
```
maybe ?

**Comment by warunicorn19** [comment]
Hey all, if you guys could please check out the lab [here]( https://www.katacoda.com/warunicorn19 ) and then let me know what all needs to be changed.

File: Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh (hunk 2 above)
Comment (suhaniii) [reply Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh:3]: I checked out https://www.katacoda.com/warunicorn19/scenarios/traefik-meshery-adapter
Following needs work :

1 . On starting lab, it is still trying to do Minikube start via /root/shell.sh
2. After installing and starting meshery this happen.
Screenshot 2021-09-01 at 8 28 24 AM

so, have to do `q`(quit)  and `y` (yes) to get 

Screenshot 2021-09-01 at 8 39 54 AM

3. Meshery dashboard shows nothing
Screenshot 2021-09-01 at 8 30 02 AM

**Comment by leecalcote** [comment]
@warunicorn19 this lab still says that it is installing minikube, despite the fact that it is not.

File: Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh (hunk 2 above)
Comment (leecalcote) [reply Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh]: > 2. After installing and starting meshery this happen.
> Screenshot 2021-09-01 at 8 28 24 AM
> 
> so, have to do `q`(quit) and `y` (yes) to get
> 
> Screenshot 2021-09-01 at 8 39 54 AM

@suhaniii we need to chase this issue down in the meshery.sh installer in `meshery/meshery`. Will you please create an issue and tag @hexxdump?

**Review by leecalcote — APPROVED** [review]
LGTM

File: Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh (hunk 2 above)
Comment (suhaniii) [reply Meshery-Adapters/traefik-meshery-adapter/step1-foreground.sh]: @leecalcote  sure

**Comment by warunicorn19** [comment]
@leecalcote I had removed that from happening, in my previous commit. 🤔

**Comment by leecalcote** [comment]
> @leecalcote I had removed that from happening, in my previous commit. 🤔

I'm not making it up (I swear) 😉

**Comment by warunicorn19** [comment]
@suhaniii is this good to merge?

**Comment by suhaniii** [comment]
@warunicorn19  Can I test same at https://katacoda.com/warunicorn19/scenarios/traefik-meshery-adapter ?

**Comment by warunicorn19** [comment]
@suhaniii umm, haven't updated it there😅.

**Comment by suhaniii** [comment]
@leecalcote  has approved and we do want to have mercy for his display from someone's [knuckles](https://github.com/layer5io/service-mesh-labs/pull/34). ☠️
We can merge it and lets resolve the issues all labs still have in next

**Comment by leecalcote** [comment]
Thank you, thank you! @warunicorn19 @suhaniii @sayantan1413

**Comment by leecalcote** [comment]
Subsequently, let's open issues for exposing on:

- layer5.io/labs
- docs.meshery.io

**Comment by warunicorn19** [comment]
@leecalcote on it.

**Pull request merged**